### PR TITLE
Restructure Database SQL code

### PR DESCRIPTION
**Pull request by MDavidson17**
Fixes: #156 

### Change Description:
SQL code has been reorganised into an easier to understand stucture
- There is a folder for each schema containing:
    - an sql file to initiate the schema and create all the tables
   - a file to insert all the default values
   - a folder containing all the functions
         - functions are sorted by the table they largely interact with

- at the top of the files a description of the functions/list of tables can be found for easy navigation through files and folders when returning to the code

### Notes for Testing:
small changes have been made to qgis-buildings-plugin to fit with some function name/schema changes
https://github.com/linz/qgis-buildings-plugin/pull/121

#### Source Code Documentation Tasks:
- [ ] README updated (where applicable)
- [ ] CHANGELOG (Unreleased section) updated
- [ ] Docstrings / comments included to help explain code

#### User Documentation Tasks:
- [ ] Confluence user guide updated (where applicable)

#### Testing Tasks:
- [ ] Added tests that fail without this change
- [ ] All tests are passing in development environment
- [ ] Reviewers assigned
- [ ] Linked to main issue for ZenHub board

### CODE REVIEW

File: scripts/nz-buildings-load [hunk 1]
```diff
@@ -0,0 +1,204 @@
+#!/bin/sh
```
Comment (dwsilk) [inline scripts/nz-buildings-load]: This shouldn't be committed, right? Just the `.in`?

File: scripts/nz-buildings-load.in [hunk 2]
```diff
@@ -101,22 +101,65 @@ psql_exit() {
     fi
 }
 
-# Execute all files in the reference sql dir
-for file in ${SCRIPTSDIR}/sql/reference/*.sql; do
+# Execute schema and table files
```
Comment (dwsilk) [inline scripts/nz-buildings-load.in]: More compact version of the next 60 lines in `sh`:

```
# Execute schema and table files

for s in buildings_common buildings buildings_reference buildings_bulk_load buildings_lds
do
    for file in ${SCRIPTSDIR}/sql/"$s"/*.sql; do
        echo ${file} >&2
        psql -f ${file} $ON_ERROR_STOP
        psql_exit
    done
done

# execute function files

for s in buildings_common buildings buildings_reference buildings_bulk_load buildings_lds
do
    for file in ${SCRIPTSDIR}/sql/"$s"/functions/*.sql; do
        echo ${file} >&2
        psql -f ${file} $ON_ERROR_STOP
        psql_exit
    done
done
```

File: makefile [hunk 3]
```diff
@@ -61,10 +90,26 @@ scripts/nz-buildings-load: scripts/nz-buildings-load.in
 install: $(SQLSCRIPTS) $(SCRIPTS_built)
 	mkdir -p ${datadir}/sql
```
Comment (dwsilk) [inline makefile]: Shorter equivalent command using `-R` (recursive):
```
install: $(SQLSCRIPTS) $(SCRIPTS_built)
	mkdir -p ${datadir}/sql
	cp -R sql/* ${datadir}/sql
	mkdir -p ${datadir}/tests/testdata
	cp tests/testdata/*.sql ${datadir}/tests/testdata
	mkdir -p ${bindir}
	cp $(SCRIPTS_built) ${bindir}
```

Yep, should have used that originally.

File: sql/buildings/02-default-values.sql [hunk 4]
```diff
@@ -71,22 +44,4 @@ INSERT INTO buildings.use (value) VALUES ('Surf Club');
...
-INSERT INTO buildings_bulk_load.bulk_load_status (value) VALUES ('Deleted During QA');
-
--- QA Status
-
-INSERT INTO buildings_bulk_load.qa_status (value) VALUES ('Not Checked');
-INSERT INTO buildings_bulk_load.qa_status (value) VALUES ('Okay');
-INSERT INTO buildings_bulk_load.qa_status (value) VALUES ('Pending');
-INSERT INTO buildings_bulk_load.qa_status (value) VALUES ('Refer to Supplier');
-INSERT INTO buildings_bulk_load.qa_status (value) VALUES ('Not Removed');
+INSERT INTO buildings.use (value) VALUES ('Water Treatment Plant');
```
Comment (dwsilk) [inline sql/buildings/02-default-values.sql]: No newline

File: scripts/nz-buildings-load (hunk 1 above)
Comment (dwsilk) [reply scripts/nz-buildings-load]: Ah, a few lines were deleted from the `.gitignore` - I think those need to be added back in with adjustments for new file locations.

File: sql/buildings/functions/04-building_outlines.sql [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+            -- integer town_city_id, integer territorial_authority_id, timestamp begin_lifespan,
+            -- shape geometry
+    --return: number of outlines updated (should only be one)
+
+-- building_outlines_update_capture_method (update capture method attribute)
+    -- params integer building_outline_id, integer capture_method_id
+    -- return: integer count number of outlines updated
+
+-- building_outlines_update_end_lifespan (update the end lifespan attr of an outline)
+    -- params: interger[]
```
Comment (YingtingChen) [inline sql/buildings/functions/04-building_outlines.sql]: integer ;)

File: sql/buildings/functions/04-building_outlines.sql [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+    SELECT count(*)::integer FROM update_capture_method;
+
+$$ LANGUAGE sql VOLATILE;
+
+COMMENT ON FUNCTION buildings.building_outlines_update_capture_method(integer, integer) IS
+'Update capture method in building_outlines table';
+
+
+-- building_outlines_update_end_lifespan (update the end lifespan attr of an outline)
+    -- params: interger[]
```
Comment (YingtingChen) [inline sql/buildings/functions/04-building_outlines.sql]: typo

File: sql/buildings/functions/07-lifecycle.sql [hunk 7]
```diff
@@ -0,0 +1,33 @@
+--------------------------------------------
+-- buildings.lifecyle
```
Comment (YingtingChen) [inline sql/buildings/functions/07-lifecycle.sql]: lifecycle

File: sql/buildings_bulk_load/01-create-schema-and-tables.sql [hunk 8]
```diff
@@ -1,18 +1,40 @@
...
+-- supplied_datasets
+-- suplied_outlines
+-- bulk_load_outlines
+-- existing_subset_extracts
+-- added
+-- removed
+-- related_groups
+-- related
+-- matched
+-- tranferred
```
Comment (YingtingChen) [inline sql/buildings_bulk_load/01-create-schema-and-tables.sql]: There're three `tranferred` in the script

File: sql/buildings_bulk_load/functions/15-compare_buildings.sql [hunk 9]
```diff
@@ -1,4 +1,16 @@
--- ADDED
+------------------------------------------------------------
+-- Script to compare all new (most recent bulk load) outlines to the
+-- existing outlines that intersect with the new outlines's capture source
```
Comment (YingtingChen) [inline sql/buildings_bulk_load/functions/15-compare_buildings.sql]: outlines'

File: sql/buildings_bulk_load/functions/14-alter_relationships_create_view.sql [hunk 10]
```diff
@@ -1,3 +1,8 @@
+------------------------------------------------------------------
```
Comment (YingtingChen) [inline sql/buildings_bulk_load/functions/14-alter_relationships_create_view.sql]: Not a function. should be put outside of the function folder

**Pull request merged**